src/SCRIPTS/BF/PAGES/pos_osd.lua

```diff
@@ -0,0 +1,250 @@
+        {["address"] = 54, ["val_1"] = 119, ["val_2"] = 120, ["name_cli"] = "osd_profile_name_pos"},
+        {["address"] = 55, ["val_1"] = 121, ["val_2"] = 122, ["name_cli"] = "osd_rssi_dbm_pos"},
+        {["address"] = 56, ["val_1"] = 123, ["val_2"] = 124, ["name_cli"] = "osd_rcchannels_pos"},
+        {["address"] = 57, ["val_1"] = 125, ["val_2"] = 126, ["name_cli"] = "osd_camera_frame_pos"},
+        {["address"] = 58, ["val_1"] = 127, ["val_2"] = 128, ["name_cli"] = "osd_efficiency_pos"},
+        {["address"] = 59, ["val_1"] = 129, ["val_2"] = 130, ["name_cli"] = "osd_total_flights_pos"},
+        {["address"] = 60, ["val_1"] = 131, ["val_2"] = 132, ["name_cli"] = "osd_up_down_reference_pos"},
+        {["address"] = 61, ["val_1"] = 133, ["val_2"] = 134, ["name_cli"] = "osd_link_tx_power_pos"},
+        {["address"] = 62, ["val_1"] = 135, ["val_2"] = 136, ["name_cli"] = "osd_wh_drawn_pos"},
+        {["address"] = 63, ["val_1"] = 137, ["val_2"] = 138, ["name_cli"] = "osd_aux_pos"},
+        {["address"] = 64, ["val_1"] = 139, ["val_2"] = 140, ["name_cli"] = "osd_ready_mode_pos"},
+        {["address"] = 65, ["val_1"] = 141, ["val_2"] = 142, ["name_cli"] = "osd_rsnr_pos"},
+        {["address"] = 66, ["val_1"] = 143, ["val_2"] = 144, ["name_cli"] = "osd_sys_goggle_voltage_pos"},
+        {["address"] = 67, ["val_1"] = 145, ["val_2"] = 146, ["name_cli"] = "osd_sys_vtx_voltage_pos"},
+        {["address"] = 68, ["val_1"] = 147, ["val_2"] = 148, ["name_cli"] = "osd_sys_bitrate_pos"},
+        {["address"] = 69, ["val_1"] = 149, ["val_2"] = 150, ["name_cli"] = "osd_sys_delay_pos"},
+        {["address"] = 70, ["val_1"] = 151, ["val_2"] = 152, ["name_cli"] = "osd_sys_distance_pos"},
+        {["address"] = 71, ["val_1"] = 153, ["val_2"] = 154, ["name_cli"] = "osd_sys_lq_pos"},
+        {["address"] = 72, ["val_1"] = 155, ["val_2"] = 156, ["name_cli"] = "osd_sys_goggle_dvr_pos"},
+        {["address"] = 73, ["val_1"] = 157, ["val_2"] = 158, ["name_cli"] = "osd_sys_vtx_dvr_pos"},
+        {["address"] = 74, ["val_1"] = 159, ["val_2"] = 160, ["name_cli"] = "osd_sys_warnings_pos"},
+        {["address"] = 75, ["val_1"] = 161, ["val_2"] = 162, ["name_cli"] = "osd_sys_vtx_temp_pos"},
+        {["address"] = 76, ["val_1"] = 163, ["val_2"] = 164, ["name_cli"] = "osd_sys_fan_speed_pos"},
+    }
+    
+if not address then
+    val_1 = items[1]["val_1"]
+    val_2 = items[1]["val_2"]
+    address = items[1]["address"]
+    name_cli = items[1]["name_cli"]
+end
+
+x = margin
+y = yMinLim - tableSpacing.header
+
+fields[#fields + 1] = { t = "OSD Element (select)", x = x, y = inc.y(lineSpacing), sp = x + sp, min = 0, max = (#items - 1), vals = { 1 }, upd = function(self) self.updateItems(self) end }
+fields[#fields + 1] = { t = "Screen",               x = x, y = inc.y(lineSpacing), sp = x + sp, min = 1, max = 1, vals = { 4 } }
+fields[#fields + 1] = { t = "CLI value (debug)",    x = x, y = inc.y(lineSpacing), sp = x + sp, min = 0, max = 999999, vals = { } }
```

Comment:
Without screen value 1 it's not working, because you have to send a screen number, I don't know if it's posible to habe it Ina a way like hidden modus?
I can test it again without screen but I remember I added it because bevor it did not save, the debug line we can remove

Reply:
It wasn't working because previously the beginning of the MSP_OSD_CONFIG payload was just overwritten with the MSP_SET_OSD_CONFIG payload. We don't have to send screen back at all, and the firmware will default it to 1. https://github.com/betaflight/betaflight/blob/b7b53399abd712476255c3b62099e04c65ba116d/src/main/msp/msp.c#L4153